src/aws-cpp-sdk-core/include/smithy/identity/auth/built-in/SigV4aAuthSchemeResolver.h

```diff
@@ -0,0 +1,25 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+
+#include <smithy/identity/auth/AuthSchemeResolverBase.h>
+#include <smithy/identity/auth/built-in/SigV4aAuthSchemeOption.h>
+
+
+namespace smithy {
+    template<typename ServiceAuthSchemeParametersT = DefaultAuthSchemeResolverParameters>
+    class SigV4aAuthSchemeResolver : public AuthSchemeResolverBase<ServiceAuthSchemeParametersT>
+    {
+    public:
+        using ServiceAuthSchemeParameters = ServiceAuthSchemeParametersT;
+        virtual ~SigV4aAuthSchemeResolver() = default;
+
+        Aws::Vector<AuthSchemeOption> resolveAuthScheme(const ServiceAuthSchemeParameters& identityProperties) override
```

Comment:
this is something i left on another review, maybe not addressed in this review -- should we be using `DefaultAuthSchemeResolverParameters` by default or another object for passing these params? seems like we dont actually initialize everything that is uses

Reply:
yes, at the least. we should add explicit constructor to fill mandatory fields
I also think that instead of letting authschemes pass different parameter types, make it pass a base polymorphic type with generic interface to fetch fields from a map of field to variant
This will unify the interface and much easier to even autogen when we have common interface signature